app/models/document.rb

```diff
@@ -222,6 +224,19 @@ def publish!
     end
   end
 
+  def upload(attachment)
```

Comment:
This doesn't feel like it belongs on the model; it's quite tightly linked to the request, since it relies on the params, so might be better in the controller.